juju/juju
# Only show local subordinates in goal state output

## Description of change

Ensure that only local subordinate units are in the "relations" section of goal state for a given unit.
The tests required a bit of surgery to swap over the unit for which the api call was being made, since mysql is the unit that could best handle having a subordinate added to it.

## QA steps

deploy a charm with multiple units
add a subordinate relation
check goal state to see that only each local subordinate unit is in goal state for any given principal

## Bug reference

https://bugs.launchpad.net/juju/+bug/1801765
